marshmallow/fields.py

```diff
@@ -401,23 +401,36 @@ def schema(self):
 
         # Inherit context from parent.
         context = getattr(self.parent, 'context', {})
+
+        # Normalize nested load_only and dump_only from parent sets
+        nested_field = '%s.' % self.name
```

Comment:
Indeed. I've made changes based on your suggestion, please let me know if it makes sense to you.